sdv/single_table/base.py

```diff
@@ -101,16 +101,16 @@ def _update_default_transformers(self):
             for sdtype, transformer in self._model_sdtype_transformers.items():
                 self._data_processor._update_transformers_by_sdtypes(sdtype, transformer)
 
-    def _check_original_metadata_updated(self):
-        if not hasattr(self, '_original_metadata'):
-            unified_metadata = Metadata.load_from_dict(self.metadata.to_dict())
-            setattr(self, '_original_metadata', unified_metadata)
+    def _check_input_metadata_updated(self):
+        if not hasattr(self, '_input_metadata'):
+            unified_metadata = Metadata.load_from_dict(self._original_metadata.to_dict())
```

Comment:
You need to check that the `_original_metadata` attribute exists for backwards compatibility.